airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -373,7 +372,8 @@ def __init__(self, kube_config, task_queue, result_queue, kube_client, worker_uu
         self.kube_client = kube_client
         self.launcher = PodLauncher(kube_client=self.kube_client)
         self.worker_configuration = WorkerConfiguration(kube_config=self.kube_config)
-        self.watcher_queue = multiprocessing.Queue()
+        self._manager = multiprocessing.Manager()
+        self.watcher_queue = self._manager.Queue()
```

Comment:
Instead of using this here could we instead use a normal Queue and use get_nowait instead? Since the only use appears to be:

```python
        while not self.watcher_queue.empty():
            self.process_watcher_task()
```

could we instead do:

```python
        try:
            while True:
                task = self.watcher_queue.get_nowait()
                self.process_watcher_task(task)
        except multiprocessing.Empty:
            pass
```

(and change process_watcher_task fn to accept the event rather than having it pull it off the queue itself).

i.e. don't ever call `.empty()` on this queue.

Reply:
I'd rather keep the current approach. In the implementation with managers, empty() has the added benefit that it will process all the events that were already put in the queue. With the get_nowait() there is a chance that some of those events already posted by other processes. Of course they will be picked up in the next heartbeat, but it somehow feels wrong. 

If you imagine that you have a busy scheduler, the get_nowait() will mean that there will pretty much always be some unprocessed entries left in the queue if you kill the scheduler. Not sure what heartbeat period usually is but it feels wrong to wait until next heartbeat. 

But I can be convinced otherwise (as I do not yet fully grasp all the details of this processing).

Reply:
Ok. I convinced myself :). Once I use manager.Queue() everywhere it's actually good idea to use get_nowait here as it will be reliable as well.

Reply:
I replaced all the loops in kubernetes executor with the same pattern now, and included the task_done() calls as those are now Joinable queues.

Reply:
Just one small difference. I prefer the case where try/except is inside the while loop and we explicitly break . It's a bit clearer what our intention is rather than pass in outer exception.